Adds welding hardhats

BuffEngineering opened the pull request:
:cl: BuffEngineering/MrDoomBringer
add: Engineers may now weld with engineering pride using the new welding hardhats!
/:cl:

Adds a hardhat that has both a light and a welding visor in the yellow, orange, white, and blue colors. The non-white variants are a bit dimmer than a normal hardhat to make them different from a hardhat and welding goggles. The CE has access to the white variant in his locker and the yellow variant is available in the EngiDrobe. Sprites have been updated, let me know if you have any suggestions.
![weldhatnew](https://user-images.githubusercontent.com/38743739/51651307-66dfbc80-1f59-11e9-821f-a5ea29f072a1.png)

Trilbyspaceclone commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
\ No newline at end of file
+	min_cold_protection_temperature = FIRE_HELM_MIN_TEMP_PROTECT
+
+/obj/item/clothing/head/hardhat/weldhat/
```
/obj/item/clothing/head/hardhat/weldhat/ -> /obj/item/clothing/head/hardhat/weldhat

Trilbyspaceclone replied:
Just remove the spare / at the end and by the ***LOOKS*** it should pass ya goof

BuffEngineering replied:
Thanks

Trilbyspaceclone replied:
Looking at it, runtiming with white one?

BuffEngineering replied:
Thanks for pointing it out, I think I got it.

subject217 commented:
nice username

tralezab commented:
Can we see the welding hardhats in game?

BuffEngineering commented:
Figured I would forget something, thanks for the reminder.

Jordie0608 commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,92 @@
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
+	toggle_helmet_light(user)
+	update_icon()
```
You're already calling `update_icon()` at the end of `toggle_helmet_light()`, no need to call again.

Jordie0608 commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,92 @@
+
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	toggle_welding_screen(user)
+	update_icon()
```
Same.

Jordie0608 reviewed (changes requested):
These seem just better, is there any reason to have a hardhat when you could have a welding hardhat?

BuffEngineering commented:
Thanks for pointing that redundancy out. The main difference is that the welding hard hat is dimmer by a tile, aside from the CE's variant, so it's a minor trade off in being able to see less dangers in general for the convenience of having a welding visor. It also lets engineers have the option to keep the engineering style without using googles that have a mad scientist feel to them.

Tlaltecuhtli commented:
the frontal sprite looks bad tho

WarJenkins commented:
i'm not sure what the front is supposed to be

MrDoomBringer commented:
@WarJenkins shit my b here u go
⠕⠓ ⠛⠕⠙ ⠕⠓ ⠋⠥⠉⠅⠕⠓ ⠛⠕⠙ ⠕⠓ ⠋⠥⠉⠅

Tlaltecuhtli commented:
why no WIP tag if the sprites are temp reeeeeee

MrDoomBringer commented:
@Tlaltecuhtli cus i was supposed to finish the sprites like an hour after the pr was posted kek

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
+	visor_flags_cover = HEADCOVERSEYES
+
+/obj/item/clothing/head/hardhat/weldhat/Initialize()
+	..()
```
```suggestion
	. = ..()
```

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
+	..()
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/user)
```
```suggestion
/obj/item/clothing/head/hardhat/weldhat/attack_self(mob/living/user)
```

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
+	update_icon()
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/user)
+	playsound(src.loc, 'sound/mecha/mechmove03.ogg', 50, 1) //Visors don't just come from nothing
```
```suggestion
	playsound(src, 'sound/mecha/mechmove03.ogg', 50, 1) //Visors don't just come from nothing
```

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
+	else
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_helmet_light(mob/user)
```
This proc contains duplicated code from `/obj/item/clothing/head/hardhat/attack_self`

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +81,90 @@
+	toggle_helmet_light(user)
+
+/obj/item/clothing/head/hardhat/weldhat/AltClick(mob/user)
+	toggle_welding_screen(user)
```
Needs ghost checks

BuffEngineering commented:
The requested changes are complete, thanks for pointing those out. Also new sprites.

Tlaltecuhtli commented:
i dont get the point tho, engineering starts with welding hats and welding goggles and after like 5 min they get welding gasmask from techwebs(tm), flashlights, pda lights and hardhat for light it seems redunant to add an hat that is a direct upgrade to all you have roundstart and is also free/roundstart

WarJenkins commented:
> 
> 
> i dont get the point tho, engineering starts with welding hats and welding goggles and after like 5 min they get welding gasmask from techwebs(tm), flashlights, pda lights and hardhat for light it seems redunant to add an hat that is a direct upgrade to all you have roundstart and is also free/roundstart
> at least give it a price tag or some shit

welding gasmasks look like total shit and i wouldn't wear them if my life depended on them
besides, it's never a bad thing to have more stuff

MrDoomBringer commented:
@Tlaltecuhtli LOTS of departments get ""straight upgrades"" to standard crew gear specifically because it's something that falls within their specialization in order to make doing their job easier than some random greyshit doing it.

Thats like complaining that botany gets free high capacity water tanks (as opposed to the crew's regular tanks), or that security gets free stunbatons (as opposed to the crew's slow+bulky stunprods), or that atmos gets a free fireproof hardsuit (as opposed to the crew's normal spacesuits), or that medbay gets free styptic spray (as opposed to the crew's generic brute patches)

Tlaltecuhtli commented:
im not talking about upgrade over standard grey shirt (now that public autolathe is gone they start with 0 welder protection except maint sunglasses) but it seems weird that you can choose between hard hat or better hard hat on the same vendor and both free, it should be a premium item at least

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -14,19 +14,27 @@
+/obj/item/clothing/head/hardhat/attack_self(mob/living/user)
+	toggle_helmet_light(user)
 
+/obj/item/clothing/head/hardhat/proc/toggle_helmet_light(mob/user)
```
```suggestion
/obj/item/clothing/head/hardhat/proc/toggle_helmet_light(mob/living/user)
```

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +89,75 @@
+	if(user.canUseTopic(src, BE_CLOSE))
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/ui_action_click(mob/user, action)
```
Do not use ui_action_click, make a custom action instead

Cyberboss commented on code/modules/clothing/head/hardhat.dm:
```diff
@@ -81,4 +89,75 @@
+	else
+		toggle_welding_screen(user)
+
+/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/user)
```
```suggestion
/obj/item/clothing/head/hardhat/weldhat/proc/toggle_welding_screen(mob/living/user)
```

FantasticFwoosh commented:
I preferred the original concept sprite, it would have justified making one out of (Plastinium Glass x1, wire x1, welder x1, screwdriver x1, hardhat x1) like the sunglasses / sunglasses huds with a deconstruction recipie if the power of it is really as weak as the OP describes with the similar reduced viewchange of a regular welding helmet.

- Now its just clear, you can't tell whether it's a faceguard or not.

I wouldn't be against a aesthetic change either for welding gas-masks this is supposed to early-game make redundant , but the function is intact.

BuffEngineering committed:
Rebase due to mime trickery

Cuts something that should be there


hardens pp


largens pp further


maximiezes pp 


fixes code


Auto stash before merge of "forth-times-the-yeah" and "BuffEngineering/forth-times-the-yeah"
Adds Initialize()


First Draft

Mostly works
Removes redundant code, adds other colors, adds to CE locker


oops forgot to


hardens pp


maximiezes pp 


Adds Initialize()


Auto stash before merge of "forth-times-the-yeah" and "BuffEngineering/forth-times-the-yeah"
oops forgot to


Attempts to balance the hardhat, adds to EngiDrobe


Removes unnecessary code and sprite, adds necessary sprite


How can commas be real if our eyes aren't real


Fixes bad slash


Fixes runtime


Optimizes code after runtime fix


Removes redundant code


Fixes 3/4


*scream


Makes visor coverage consistent with new sprites


Actually fixes the issue


samus? its an honor


Makes requested changes


Doesn't delete welding hardhat sprites
